datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -1615,6 +1659,91 @@ mod tests {
         assert_optimized_plan_eq(plan, expected)
     }
 
+    #[test]
+    fn distinct_on() -> Result<()> {
```

Comment:
I think we also need a "false negative" test here, i.e. a filter on the same subquery as the distinct (so without an `alias`) but not on a distinct column.

Something like (not tested):
```rust
    fn distinct_on_() -> Result<()> {
        let table_scan = test_table_scan()?;
        let plan = LogicalPlanBuilder::from(table_scan)
            .distinct_on(vec![col("a")], vec![col("a"), col("b")], None)?
            .filter(col("b").eq(lit(1i64)))?
            .build()?;
        // filter is on the same subquery as the distinct, so it should be pushed down even though it isn't in the ON clause
        let expected = "\
        DistinctOn: on_expr=[[test.a]], select_expr=[[a, b]], sort_expr=[[]]\
        \n  TableScan: test, full_filters=[b = Int64(1)]";
        assert_optimized_plan_eq(plan, expected)
    }
```